src/search/pdbs/pattern_generator_rcg.cc

```diff
@@ -0,0 +1,76 @@
+#include "pattern_generator_rcg.h"
+
+#include "pattern_database.h"
+#include "rcg.h"
+#include "utils.h"
+
+#include "../option_parser.h"
+#include "../plugin.h"
+#include "../task_proxy.h"
+
+#include "../utils/logging.h"
+#include "../utils/rng.h"
+#include "../utils/rng_options.h"
+#include "../utils/timer.h"
+
+#include <vector>
+
+using namespace std;
+
+namespace pdbs {
+
+PatternGeneratorRCG::PatternGeneratorRCG(
+    options::Options& opts)
+    : max_pdb_size(opts.get<int>("max_pdb_size")),
+      max_time(opts.get<double>("max_time")),
+      bidirectional(opts.get<bool>("bidirectional")),
+      verbosity(opts.get<utils::Verbosity>("verbosity")),
+      rng(utils::parse_rng_from_options(opts)) {
+}
+
+PatternInformation PatternGeneratorRCG::generate(
+    const std::shared_ptr<AbstractTask> &task) {
+    utils::Timer timer;
+    if (verbosity >= utils::Verbosity::NORMAL) {
+        utils::g_log << "Single RCG: generating patterns" << endl;
+    }
+    vector<vector<int>> cg_neighbors = compute_cg_neighbors(
+        task, bidirectional);
+    TaskProxy task_proxy(*task);
+    vector<FactPair> goals = get_goals_in_random_order(task_proxy, *rng);
+
+    Pattern pattern = generate_pattern_rcg(
+        max_pdb_size,
```

Comment:
I actually like this formatting but I think it doesn't follow our conventions.

Reply:
I'm also not sure, but I also don't see that they would directly violate them: http://www.fast-downward.org/ForDevelopers/Whitespace
There is no example on what happens when the function call is too large to fit into one line even when breaking after the opening parenthesis. I used this style you see here not only here...

Reply:
I interpreted the first example to continue like this:

```
int foobar1 = example_of_breaking_after_parenthesis(
    some_argument, foo + bar, baz, more_arguments,
    even_more_arguments, final_argument);
```

but you're right, it doesn't explicitly say so. I'm fine with this anyway, I like having one parameter per line so I don't want to argue too hard against it.